Data/Root/RootDataIO.cpp

```diff
@@ -78,7 +77,7 @@ RootDataIO::readData(const std::string &InputFilePath) const {
 
     Events.push_back(evt);
   } // end event loop
-  File.Close();
+  delete chain;
```

Comment:
Root will not try to delete this again? You could also try to use TChain like this `TChain chain(...);` to avoid the new in the first place.

Reply:
Hehe, it looks terrible, I know. But this is ROOT... for large files, you need to allocate it dynamically. Actually, there will be a warning message at run-time if you don't (let me look up some documentation on that issue). A `unique_ptr` is a small remedy in this case.

And yes, `delete` is fine with ROOT, it won't delete again.

Reply:
Ok. Yeah it would be fine to use pointers here. I just remember I had troubles sometimes when I deleted the pointer myself, because root's garbage collection tried to do that again at the end of the program...